modules/standard/Map.chpl

```diff
@@ -95,7 +95,7 @@ module Map {
       :arg parSafe: If `true`, this map will use parallel safe operations.
       :type parSafe: bool
     */
-    proc init=(const ref other: map(?kt, ?vt, ?ps)) {
+    proc init=(ref other: map(?kt, ?vt, ?ps)) {
```

Comment:
This might be an improvement today, but doesn't it make it impossible to copy-init from a map declared as `const myMap: map(...)` ? Should instead there be two `init=` - one uses a `ref` argument and one does not - and a `where` clause to choose which one is appropriate? Also does this suggest that we need `ref-if-modified` to be a more generally available user feature?

You could adjust the implementation to use INTENT_REF_MAYBE_CONST intent on `other` here (probably with a pragma applied to the argument). However I think it'd be reasonable to propose something like `const? ref other: map` as syntax to indicate INTENT_REF_MAYBE_CONST.

Reply:
It seems like `ref-if-modified` might be generally useful to make things work with managed classes.

I tried adding `pragma "default intent is ref if modified"` to `other`, but I'm getting illegal lvalue errors on the `vals[key] = other.vals[key];` line. The actual and formal it is complaining about both have type `_ref(_owned(borrowed C?))`.

Reply:
I would expect that pragma to only apply to a type. I think you'd have to add a different pragma/flag and wire it up to INTENT_REF_MAYBE_CONST.

Reply:
OK, thanks. That looks like it worked.